### PR TITLE
[8.0] server environment/mail_environment dont error out on missing imports

### LINKED ISSUE #1546: [8.0] ImportError: No module named server_environment_files
The last commits on repo re-create the issue on server_environment_files import, the problem is not present at 5ec40c3e47e8ab0fa090b4f51237a8673ce11d85

Error is ```ImportError: No module named server_environment_files``` as #89

### PR DESCRIPTION
**Pull request by thomaspaulb**
Fixes #1546

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

**Comment by thomaspaulb** [comment]
@sergiocorato

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:

A forward port to 10.0 would be great

**Review by sergiocorato — APPROVED** [review]

**Comment by sergiocorato** [comment]
@thomaspaulb @pedrobaeza thanks!

**Comment by pedrobaeza** [comment]
Pylint was already failing before, so I merge this.

**Comment by thomaspaulb** [comment]
> A forward port to 10.0 would be great

@aheficent [10.0](https://github.com/OCA/server-tools/commit/ff3d8679) and [11.0](https://github.com/OCA/server-env/commit/d9ad47f40d3e5657b0112b7a877e1cc83569808a) already seem to have the same.

File: mail_environment/__init__.py [hunk 1]
```diff
@@ -1,2 +1,10 @@
...
-from . import env_mail
+import logging
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from . import env_mail
+except ImportError:
+    _logger.info("ImportError raised while loading module.")
+    _logger.debug("ImportError details:", exc_info=True)
```
Comment (sbidoul) [inline mail_environment/__init__.py:10]: Why protecting a local import?

**Comment by sbidoul** [comment]
@pedrobaeza At first glance this PR does NOT look good to me. It is too complex with many useless import protections. It should be redone the same way it is done in 10 which is much cleaner.

**Comment by pedrobaeza** [comment]
Well, it was done working successfully in v9 and this was a blocking point for some of the contributors, so I acted quickly. You can propose a better alternative if you want.

**Comment by yucer** [comment]
Hello. Don't you get this error when the server-tools repo stays in the *addons-path* ?

I saw that in an old v8 instance, maybe it happens also for v9 ?

```
ImportError: No module named server_environment_files
```

**Comment by yucer** [comment]
Sorry I guess it was a bad setup.

**Pull request merged**